plugins/woocommerce-blocks/assets/js/blocks/product-collection/constants.ts

```diff
@@ -71,7 +71,7 @@ export const DEFAULT_ATTRIBUTES: Partial< ProductCollectionAttributes > = {
 		columns: 3,
 		shrinkColumns: true,
 	},
-	queryContextIncludes: [ 'collection' ],
+	queryContextIncludes: [ 'collection', 'previewState' ],
```

Comment:
Given that attributes aren't needed here, then this should be passed "hard-coded" into the rest_product_query context.